packages/fields/src/fields/date/date-view.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { dateI18n, getDate, getSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import type { BasePost } from '../../types';
+
+const getFormattedDate = ( dateToDisplay: string | null ) =>
+	dateI18n(
```

Comment:
One interesting thing from migrating this to TypeScript is that it forces you to pass the last parameter. Despite what the [docs say](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n), [there's no default value and cannot be undefined](https://github.com/WordPress/gutenberg/blob/move/date-field/packages/date/src/index.js#L543).

Reply:
I think this can be solved by slightly changing the type definition of the function and making the argument optional `=something` rather than `something | undefined`

Reply:
Reverted the default parameter and modified the JSDoc instead at https://github.com/WordPress/gutenberg/pull/66938/commits/544bf3e3d60bfdf953eab4813b70595a693886fb